src/main/java/hudson/tasks/junit/pipeline/JUnitResultsStepExecution.java

```diff
@@ -48,7 +52,14 @@ protected TestResultSummary run() throws Exception {
         pipelineTestDetails.setNodeId(nodeId);
         pipelineTestDetails.setEnclosingBlocks(getEnclosingBlockIds(enclosingBlocks));
         pipelineTestDetails.setEnclosingBlockNames(getEnclosingBlockNames(enclosingBlocks));
+
         try {
+            // If we are within a withChecks context, and have not provided a name override in the step, apply the withChecks name
+            if (Util.fixEmpty(step.getChecksName()) == null) {
+                Optional.ofNullable(getContext().get(ChecksInfo.class))
+                        .map(ChecksInfo::getName)
+                        .ifPresent(step::setChecksName);
```

Comment:
I was wondering if we should be deprecating the explicit setting of the checks name?

going forward I wouldn't expect we would need to add these overrides to steps as we have `withChecks` what do you think?

Reply:
It makes a nice shorthand - for instance

```groovy
withChecks('Maven build') {
    sh 'maven clean install'
    publishChecks()
}
junit testResults: '*.xml', checksName: 'Custom tests name'
```

rather than

```groovy
withChecks('Maven build') {
    sh 'maven clean install'
    publishChecks()
}
withChecks('Custom tests name') {
    junit testResults: '*.xml'
}
```

but to be honest, that second one doesn't look to bad, and I'm struggling to find a use case where you'd actually want to override a `withChecks` setting. I certainly can't think of one within the suite of Jenkinsfiles I manage.

Reply:
I guess it's needed for declarative pipeline? the override that is, until we ship a compatible version for it.

unless withChecks already works for that?

Reply:
Apparently it Just Works (tm)  - https://github.com/jenkinsci/junit-plugin/pull/231/commits/c2e320810de425f726082a7494a2e0225845505c

Reply:
hmm ok, I guess we leave it in for now although it does cause this somehow:
https://github.com/jenkinsci/junit-plugin/issues/222

Which I haven't managed to figure it out

Reply:
Was that caused by https://github.com/jenkinsci/junit-plugin/pull/199/files#diff-4fa8f2162a640cc3f024dadbffc9b442ab73e2afd30e5030ec0088ac300371e5L78 ?

Reply:
ah yeah probably, I guess it'll go away as jobs are re-configured? unless we want to add the field back, bit annoying though